### PR TITLE
[tests] Fix various spec tests

### PR DESCRIPTION
**Pull request by gahaas**
This PR fixes some issues in the spec tests:
* Add missing functions to the default import object of the spec tests;
* Adjust the byte encoding of a test in the binary-leb128.wast to test multi-byte table indices in 
  element segments.

### CODE REVIEW

File: test/core/ref_func.wast [hunk 1]
```diff
@@ -93,6 +93,7 @@
   (elem (table $t) (i32.const 0) funcref (ref.func $f4))
   (elem func $f5)
   (elem funcref (ref.func $f6))
+  (elem declare func $f1 $f2 $f3 $f4 $f5 $f6)
```
Comment (rossberg) [inline test/core/ref_func.wast:96]: The point of this test is that this is no longer needed, per the resolution of #76. The occurrences in the other declarations are sufficient.

File: test/core/table.wast [hunk 2]
```diff
@@ -6,7 +6,7 @@
 (module (table 0 1 funcref))
 (module (table 1 256 funcref))
 (module (table 0 65536 funcref))
-(module (table 0 0xffff_ffff funcref))
+(module (table 0 0x8f_ffff funcref))
```
Comment (rossberg) [inline test/core/table.wast:9]: The limitation you cite concerns the concrete (and web-specific!) _dynamic_ limit, not the static one. So it merely means that you cannot successfully grow beyond that. It is still a valid module -- validity must not be environment-dependent.

File: test/core/ref_func.wast (hunk 1 above)
Comment (gahaas) [reply test/core/ref_func.wast]: Ah, I missed that. Thanks! I removed the change.

File: test/core/table.wast (hunk 2 above)
Comment (gahaas) [reply test/core/table.wast:9]: As far as I understand the spec text there, this is not a dynamic limit. The text says: 

> An implementation must reject a module that exceeds these limits with a {{CompileError}}.

Comment (rossberg) [reply test/core/table.wast:9]: Hm, that's an oversight/bug in the JS API spec then. The core spec explicitly lists table size under [execution limitations](https://webassembly.github.io/spec/core/appendix/implementation.html#execution).

Comment (gahaas) [reply test/core/table.wast:9]: I created a PR now that resolves that inconsistency, see https://github.com/WebAssembly/spec/pull/1195. Can we change the value here until the inconsistency in the spec is resolved?

Comment (rossberg) [reply test/core/table.wast:9]: Thanks for PR! It seems like we should be able to land it today, so maybe we can defer this PR for a few hours and avoid a temporary hack? :)

Comment (gahaas) [reply test/core/table.wast]: Ack. I reverted the change to this test.

**Pull request merged**